giskard/llm/talk/config.py

```diff
@@ -0,0 +1,154 @@
+    "rows from the dataset, then you run the model prediction on that rows and "
+    "finally return the prediction result."
+)
+
+_PREDICT_USER_INPUT_TOOL_DESCRIPTION = (
+    "Your task is to return prediction of a feature vector (partially) built from "
+    "the user input. You expect a dictionary with features and their values, "
+    "extracted from the user input, then you fill the values of features not "
+    "mentioned in the user query and build input row. Then you run the model "
+    "prediction on that row and finally return the prediction result."
+)
+
+_PREDICT_TOOL_DESCRIPTION = (
+    "Your task is to return prediction of a feature vector either extracted from"
+    "the dataset, or partially built from the user input. You expect a dictionary "
+    "with features and their values. First you try to filter rows from the dataset. "
+    "If it is not possible, you fill the values of features not mentioned in "
+    "the user query to build an input vector. Then you run the model prediction on "
+    "that vector and finally return the prediction result."
+)
+
+_CALCULATE_METRIC_TOOL_DESCRIPTION = (
+    "Your task is to calculate a performance metric either for a classification or a "
+    "regression model on a provided dataset. You expect two parameters: a metric name, "
+    "and a dictionary with features and their values. First, you filter rows from the "
+    "dataset, if it is necessary. Then you run the model prediction on that rows or on "
+    "the whole dataset to get the prediction result. Finally, based on obtained "
+    "predictions and the ground truth labels of the dataset, you calculate the value of "
+    "a chosen performance metric and return it."
+)
+
+_SHAP_EXPLANATION_TOOL_DESCRIPTION = (
+    "You expect a dictionary with feature names as keys and their values as dict "
+    "values, which you use to filter rows in the dataset, then you run the SHAP "
+    "explanation on that filtered rows, and finally you return the SHAP explanation "
+    "result as well as the model prediction result."
+    "Please note, that the bigger SHAP value - the more important feature is for the"
+    "prediction."
+)
+
+_ISSUES_SCANNER_TOOL_DESCRIPTION = (
+    "Your task is to give the user a summary of the 'giskard' scan result. For your "
+    "info, this feature is used to detect ML model's vulnerabilities such as "
+    "performance bias, hallucination, prompt injection, data leakage, spurious "
+    "correlation, overconfidence, etc. You return a table, where the important "
+    "columns are 'Description', 'Vulnerability', 'Level', 'Metric' and 'Deviation'"
+)
+
+
+class ToolDescription(str, Enum):
+    PREDICT = _PREDICT_TOOL_DESCRIPTION
+    CALCULATE_METRIC = _CALCULATE_METRIC_TOOL_DESCRIPTION
+    SHAP_EXPLANATION = _SHAP_EXPLANATION_TOOL_DESCRIPTION
+    ISSUES_SCANNER = _ISSUES_SCANNER_TOOL_DESCRIPTION
+
+
+AVAILABLE_METRICS = {
+    "accuracy": classification_report,
```

Comment:
I am using the classification report, because it allows to get necessary performance metrics (accuracy, f1, precision, recall) per each label, with just one function call. Otherwise I would have to do additional filtering by classification label for f1, precision, recall. `classification_report` allows to simplify metrics calculation.